### DISCUSSION TITLE
[Feature Request] Add methods for manipulating block.timestamp and block.number in local test node

### DISCUSSION THREAD

**Discussion by smusialpc** [zkSync-Community-Hub/zksync-developers#77] [Feature requests]
### Network

zkSync Era

### Provide a brief description of the feature you would like to see implemented.

It would be useful to have methods that allow setting block timestamp or number of next mined block in test in-memory node. Such functionality would be useful in testing time-dependent smart contracts.

Functionality already exists in hardhat:
- setNextBlockTimestamp(timestamp) -> hardhat-network-helpers plugin
- evm_setNextBlockTimestamp -> JSON-RPC

### Who does this feature request benefit the most (users, developers, protocol, etc.)?

Developers

**Comment by uF4No** [comment]
Noted! This is one of the features we're working on for the [in-memory node](https://era.zksync.io/docs/tools/testing/era-test-node.html) which we will integrate with Hardhat to run tests locally. We'll announce it once ready

**Reply by MexicanAce** [reply]
GitHub Issue created in the `era-test-node` repo : https://github.com/matter-labs/era-test-node/issues/29

Resolving this comment as the answer for this discussion, and moving further conversation to the Issue above ☝️

**Comment by bricnounlerumark** [comment]
This feature request, which enables setting block timestamp or number of next mined block in the zkSync Era test in-memory node, primarily benefits developers who are testing time-dependent smart contracts. It would enhance the testing process by providing more control over block timestamps and advancing the blockchain state for accurate simulation of time-dependent scenarios.

**Comment by whaliop** [comment]
Requesting the implementation of methods to set block timestamp or number in the zkSync Era test in-memory node for testing time-dependent smart contracts.

**Comment by eleykelly** [comment]
Feature Request: Implement methods to set block timestamp or number of next mined block in zkSync Era test in-memory node for testing time-dependent smart contracts.
Beneficiaries: Developers

**Comment by latoyatidwell** [comment]
Adding methods to set block timestamp or the number of the next mined block in the test in-memory node would greatly benefit developers, especially when testing time-dependent smart contracts.

**Comment by dutterbutter** [comment]
This is now supported in era-test-node, please refer to the this [tests file](https://github.com/matter-labs/era-test-node/blob/main/e2e-tests/test/evm-apis.test.ts) that illustrates usage well. Closing this request as delivered.

**Comment by ngopriyanka** [comment]
This feature request proposes adding methods to the zkSync Era network that allow setting the block timestamp or number of the next mined block in a test in-memory node. This functionality would be useful for testing time-dependent smart contracts. Developers would benefit the most from this feature. The functionality already exists in hardhat through the setNextBlockTimestamp method in the hardhat-network-helpers plugin and the evm_setNextBlockTimestamp JSON-RPC method.
